src/map/clif.c

```diff
@@ -19104,6 +19105,478 @@ unsigned short clif_parse_cmd_optional( int fd, struct map_session_data *sd ) {
 }
 
 /*==========================================
+ * RoDEX
+ *------------------------------------------*/
+
+void clif_parse_rodex_open_write_mail(int fd, struct map_session_data *sd)
+{
+	const struct PACKET_CZ_REQ_OPEN_WRITE_MAIL *rPacket = RFIFOP(fd, 0);
+	int8 result = (rodex->isenabled() == true) ? 1 : 0;
+	
+	clif->rodex_open_write_mail(fd, rPacket->receiveName, result);
+}
+
+void clif_rodex_open_write_mail(int fd, const char receiver_name[NAME_LENGTH], int8 result)
+{
+#if PACKETVER >= 20140416
+	struct PACKET_ZC_ACK_OPEN_WRITE_MAIL *sPacket = NULL;
+	WFIFOHEAD(fd, sizeof(*sPacket));
+	sPacket = WFIFOP(fd, 0);
+	sPacket->PacketType = rodexopenwrite;
+	strncpy(sPacket->receiveName, receiver_name, sizeof(sPacket->receiveName));
+	sPacket->result = result;
+	WFIFOSET(fd, sizeof(*sPacket));
+#endif
+}
+
+void clif_parse_rodex_add_item(int fd, struct map_session_data *sd)
+{
+	const struct PACKET_CZ_ADD_ITEM_TO_MAIL *rPacket = RFIFOP(fd, 0);
+	int16 idx = rPacket->index - 2;
+
+	nullpo_retv(sd);
+
+	rodex->add_item(sd, idx, (int16)rPacket->count);
+}
+
+void clif_rodex_add_item_result(struct map_session_data *sd, int16 idx, int16 amount, int8 result) {
+#if PACKETVER >= 20140416
+	struct PACKET_ZC_ADD_ITEM_TO_MAIL *packet;
+	int fd, j;
+
+	nullpo_retv(sd);
+
```

Comment:
hmm, this is a result function and it can be called with the invalid idx when an invalid idx is received on rodex.c, in this case with an error result code, do I need to check it there too? (you can see what I'm saying in rodex.c at line 86, inside rodex_add_item)